cinema/serializers.py

```diff
@@ -1,6 +1,59 @@
 from rest_framework import serializers
+from cinema.models import Movie, Genre, Actor, CinemaHall
 
-from cinema.models import Movie
+
+class GenreSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=255)
+
+    def create(self, validated_data):
+        return Genre.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.name = validated_data.get("name", instance.name)
+        instance.save()
+        return instance
+
+
+class ActorSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    first_name = serializers.CharField(max_length=255)
+    last_name = serializers.CharField(max_length=255)
+
+    def create(self, validated_data):
+        return Actor.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.first_name = validated_data.get(
+            "first_name",
+            instance.first_name
+        )
+        instance.last_name = validated_data.get(
+            "last_name",
+            instance.last_name
+        )
+        instance.save()
+        return instance
+
+
+class CinemaHallSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    name = serializers.CharField(max_length=255)
+    rows = serializers.IntegerField()
+    seats_in_row = serializers.IntegerField()
+
+    def create(self, validated_data):
+        return CinemaHall.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.name = validated_data.get("name", instance.name)
+        instance.rows = validated_data.get("rows", instance.rows)
+        instance.seats_in_row = validated_data.get(
+            "seats_in_row",
+            instance.seats_in_row
+        )
+        instance.save()
+        return instance
 
 
 class MovieSerializer(serializers.Serializer):
```

Comment:
The `MovieSerializer` is missing fields for `actors` and `genres`. Since these are `ManyToMany` relationships in the `Movie` model, they should be included in the serializer. Consider using `PrimaryKeyRelatedField` or `StringRelatedField` with `many=True` to handle these relationships.